.github/workflows/deploy_lambda.yml

```diff
@@ -0,0 +1,182 @@
+name: Deploy Lambda Functions
+on:
+  workflow_call:
+    inputs:
+      environment: # development | release | production
+        required: true
+        type: string
+      rollback: # In case of rollback
+        required: false
+        default: false
+        type: boolean
+
+    secrets:
+      GIT_CREDENTIALS:
+        required: true
+      NPM_AUTH_TOKEN:
+        required: true
+      AWS_OPS_BUCKET:
+        required: true
+      AWS_ACCESS_KEY_ID:
+        required: true
+      AWS_SECRET_ACCESS_KEY:
+        required: true
+      AWS_ROLE:
+        required: true
+      PRIVATE_KEY:
+        required: true
+
+jobs:
+  changes:
+    name: 'Check Changes'
+    if: ${{ !inputs.rollback }}
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          ref: ${{ github.ref_name }}
+          fetch-depth: 5
+      - uses: marceloprado/has-changed-path@v1
+        id: terraform_changed
+        with:
+          paths: terraform
+      - uses: marceloprado/has-changed-path@v1
+        id: code_changed
+        with:
+          paths: api consumers shared/common shared/config
+      - run: echo Has any TF code been updated? - ${{ steps.terraform_changed.outputs.changed }}
+      - run: echo Has any code been updated? - ${{ steps.code_changed.outputs.changed }}
+    outputs:
+      # terraform_changed: ${{ steps.terraform_changed.outputs.changed }}
+      # code_changed: ${{ steps.code_changed.outputs.changed }}
+      terraform_changed: true
+      code_changed: true
+
+  temp_vars:
+    name: 'Define Temporary Variables'
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set short env
+        id: short_env
+        run: |
+          case ${{ inputs.environment }} in
+            development) echo "::set-output name=short_env::dev" ;;
+            release) echo "::set-output name=short_env::release" ;;
+            production) echo "::set-output name=short_env::prod" ;;
+            *) echo "Invalid environment" && exit 1 ;;
+          esac
```

Comment:
In case of the environment var is an invalid environment, it will fail the workflow?
IMO it should fail and do not run the deploy

Reply:
yeah!
`exit 1`  should ensure that the workflow fails.